File: src/com/boardgamegeek/provider/BuddiesProvider.java
package com.boardgamegeek.provider;

import android.content.ContentValues;
import android.database.sqlite.SQLiteDatabase;
import android.net.Uri;

import com.boardgamegeek.provider.BggContract.Buddies;
import com.boardgamegeek.provider.BggDatabase.Tables;
import com.boardgamegeek.util.SelectionBuilder;

public class BuddiesProvider extends BaseProvider {
	private static final String TABLE = Tables.BUDDIES;

	@Override
	protected SelectionBuilder buildSimpleSelection(Uri uri) {
		return new SelectionBuilder().table(TABLE);
	}

	@Override
	protected String getPath() {
		return "buddies";
	}

	@Override
	protected String getType(Uri uri) {
		return Buddies.CONTENT_TYPE;
	}

	@Override
	protected Uri insert(SQLiteDatabase db, Uri uri, ContentValues values) {
		insert(db, uri, values, TABLE);
		return Buddies.buildBuddyUri(values.getAsInteger(Buddies.BUDDY_ID));
	}

}

File: app/src/main/java/com/boardgamegeek/provider/BuddiesProvider.kt
package com.boardgamegeek.provider

import android.content.ContentValues
import android.net.Uri
import com.boardgamegeek.provider.BggContract.Buddies
import com.boardgamegeek.provider.BggContract.PATH_BUDDIES
import com.boardgamegeek.provider.BggDatabase.Tables

class BuddiesProvider : BasicProvider() {
    override fun getType(uri: Uri) = Buddies.CONTENT_TYPE

    override val path: String = PATH_BUDDIES

    override val table = Tables.BUDDIES

    override val defaultSortOrder = Buddies.DEFAULT_SORT

    override fun insertedUri(values: ContentValues?, rowId: Long): Uri? {
        val buddyName = values?.getAsString(Buddies.BUDDY_NAME)
        return if (buddyName.isNullOrBlank()) null else Buddies.buildBuddyUri(buddyName)
    }
}